### PR TITLE
Fix missing linker language for SIMD

### PR DESCRIPTION
**Pull request by crtrott**
In algorithms we worked around that with a dummy cpp file:

Error:
```
CMake Error: CMake can not determine linker language for target: kokkossimd
CMake Error in kokkos/simd/src/CMakeLists.txt:
  Exporting the target "kokkossimd" is not allowed since its linker language
  cannot be determined
```

### CODE REVIEW

**Review by dalg24 — APPROVED** [review]
For the record that is an alternative to the `dummy.cpp` file in algorithms.
I also like it better than the "dummy" source file.

File: CMakeLists.txt [hunk 1]
```diff
@@ -67,7 +67,13 @@ ENDFUNCTION()
 
 LIST(APPEND CMAKE_MODULE_PATH cmake/Modules)
 
-IF(NOT KOKKOS_HAS_TRILINOS)
+
+IF(KOKKOS_HAS_TRILINOS)
+  # What language are we compiling Kokkos as
+  # downstream dependencies need to match this!
+  # We do not supporting using CUDA or HIP as cmake language yet in Trilinos
+  SET(KOKKOS_COMPILE_LANGUAGE CXX)
```
Comment (PhilMiller) [inline CMakeLists.txt:75]: It looks like this appears in both the `IF` and `ELSE` branches - why not lift it up above?

Comment (crtrott) [reply CMakeLists.txt:75]: Actually it gets reset again in the else branch if you have CUDA on, I thought its cleaner if the Trilinos path makes clear we always set it to CXX and it doesn't get messed with later again.

Comment (PhilMiller) [reply CMakeLists.txt:75]: Works for me

**Review by PhilMiller — APPROVED** [review]

**Comment by PhilMiller** [comment]
Given the CMake failures in Jenkins, you may have to do the `dummy.cpp` thing

File: CMakeLists.txt [hunk 2]
```diff
@@ -67,7 +67,13 @@ ENDFUNCTION()
 
 LIST(APPEND CMAKE_MODULE_PATH cmake/Modules)
 
-IF(NOT KOKKOS_HAS_TRILINOS)
+
+IF(KOKKOS_HAS_TRILINOS)
+  # What language are we compiling Kokkos as
+  # downstream dependencies need to match this!
+  # We do not supporting using CUDA or HIP as cmake language yet in Trilinos
```
Comment (dalg24) [inline CMakeLists.txt:74]:
```suggestion
  # We do not support using CUDA or HIP as CMake language yet in Trilinos
```

**Pull request merged**